cw8.c

```diff
@@ -0,0 +1,227 @@
+            }
+        }
+        det += (x % 2 == 0 ? 1 : -1) * matrix[0][x] * determinant(n - 1, submatrix);
+    }
+
+    return det;
+}
+
+int main() {
+    int choice;
+    printf("Select a task (1-7): ");
+    scanf("%d", &choice);
+
+    switch (choice) {
+        case 1: {
+            int matrix1[3][3] = {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}};
+            int N, M;
+            printf("Task 1: Enter N and M: ");
+            scanf("%d %d", &N, &M);
+            task_1(matrix1, N, M);
+            print_matrix(3, 3, (double (*)[3])matrix1, 1); // виводимо як цілу матрицю
+            break;
+        }
+
+        case 2: {
+            double matrix2[3][3] = {{1.0, 2, 3}, {4, 5, 6}, {7, 8, 9}};
+            int i, j;
+            double a;
+            printf("Task 2: Enter i, j and a: ");
+            scanf("%d %d %lf", &i, &j, &a);
+            task_2(matrix2, i, j, a);
+            print_matrix(3, 3, matrix2, 0); // виводимо як дійсну матрицю
+            break;
+        }
+
+        case 3: {
+            int m, n;
+            printf("Task 3: Enter m and n: ");
+            scanf("%d %d", &m, &n);
+            double matrix3[m][n];
+            input_matrix(m, n, matrix3);
+            print_matrix(m, n, matrix3, 0);
+            break;
+        }
+
+        case 4: {
+            int m, n;
+            printf("Task 4: Enter m and n: ");
+            scanf("%d %d", &m, &n);
+            double matrix4[m][n];
+            task_4(m, n, matrix4);
+            print_matrix(m, n, matrix4, 0);
+            break;
+        }
+
+        case 5: {
+            double matrix5[3][3] = {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}}, matrix_transpose[3][3];
+            transpose(3, matrix5, matrix_transpose);
+            print_matrix(3, 3, matrix_transpose, 0);
+            break;
+        }
+
+        case 6: {  // 6: Сума елементів, де i - j = k
+            unsigned m, n;
+            int k;
+
+            double sum = 0.0;
+            printf("Enter number (i - j) = k: \n");
+            scanf("%d", &k);
+            printf("Enter m and n : \n");
+            scanf("%u %u", &m, &n);
+
+            double matrix[m][n];
+
+            for(int i = 0; i < m; i++) {  // if (m <= MAX_LARGE && n <= MAX_LARGE)
+                printf("matrix[%d]: \n", i);
+                for(int j = 0; j < n; j++) {
+                    scanf("%lf", &matrix[i][j]);
+                }
+            }
+            for(int i = 0; i < m; i++) {
+                for(int j = 0; j < n; j++) {
+                    if ((i - j) == k) {
+                        sum += matrix[i][j];
+                    }
+                }
+            }
+            printf("The sum is: %.2lf\n", sum);  /*else {
+            printf("m and n should be <= %d!\n", MAX_LARGE);*/
+        }  // return sum;
+        // }
+
+        case 7: {
+            int n;
+            printf("Task 7: Enter n (the size of the square matrix): ");
+            scanf("%d", &n);
+            double matrix7[n][n];

```

Comment:
Not cross platforming

Reply:
Fixed dynamic declaration of arrays based on input sizes.